lib/rules/unit-no-unknown/__tests__/index.js

```diff
@@ -220,6 +220,22 @@ testRule(rule, {
 			code: "@import 'foo.css'",
 			description: 'ignore non-media and non-variable at-rule',
 		},
+		{
+			code: "a { background-image: image-set('img-1x.jpg' 1x, 'img-2x.jpg' 2x, 'img-3x.jpg' 3x) }",
+			description: 'ignore `x` unit in image-set',
+		},
+		{
+			code: '@media (resolution: 2x) {}',
+			description: 'ignore `x` unit in @media with `resolution`',
+		},
+		{
+			code: '@media ( resOLution:  2x) {}',
+			description: 'ignore `x` unit in @media with `resolution`',
+		},
+		{
+			code: 'a { image-resolution: 1x; }',
+			description: 'ignore `x` unit in image-resolution',
+		},
```

Comment:
Let's add test with `@media (resolution: 2x) and (min-width: 200x) {}`, second should be rejected

Reply:
I had to update logic so that such a case could be handled properly.
Not sure about implementation though.
Any suggestions are welcome.